src/main/java/org/embulk/filter/column/ColumnFilterPlugin.java

```diff
@@ -117,7 +120,7 @@ static Schema buildOutputSchema(PluginTask task, Schema inputSchema)
                 boolean matched = false;
                 for (ColumnConfig dropColumn : dropColumns) {
                     // skip json path notation to build outputSchema
-                    if (dropColumn.getName().startsWith("$.")) {
+                    if (PathCompiler.isDocContext(dropColumn.getName())) {
```

Comment:
https://github.com/kysnm/JsonPathCompiler/blob/6539a340638ddf2d71912c1ca28f52083a944a2d/src/main/java/com/github/kysnm/jsonpathcompiler/expressions/path/PathCompiler.java#L48-L53

```
    public static Boolean isStartsWithDoller    (String path)
    {
        StringBuilder dotNotationRootPath = new StringBuilder(DOC_CONTEXT).append(PERIOD);
        StringBuilder bracketNotationRootPath = new StringBuilder(DOC_CONTEXT).append(OPEN_SQUARE_BRACKET);
        return path.contains(dotNotationRootPath.toString()) || path.contains(bracketNotationRootPath.toString());
    }
```

is wrong. 1. It must use `startsWith`, 2. The name does not describe what it is doing.